### PR TITLE
add support for building with meson

### PR DESCRIPTION
**Pull request by The-King-of-Toasters**
This is just #80 again because I screwed up rebasing, thanks git.

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -0,0 +1,208 @@
+project(
+        'htop',
+	'c',
```
Comment (hugomg) [inline meson.build:3]: It looks like it is mixing tabs and spaces here?

Comment (The-King-of-Toasters) [reply meson.build:3]: oops, lemme fix that.

**Comment by The-King-of-Toasters** [comment]
I need to go through the last couple commits and see if I need to change anything (e.g. HTOP_LINUX et al.).

**Comment by The-King-of-Toasters** [comment]
Ok, I made hwloc and delay accounting a feature, so that if they're disabled at build time they won't be included in the lib list. Also added checks for linux-affinity and hwloc, along with adding the header for setuid support. Will test on other machines.

**Comment by BenBE** [comment]
@The-King-of-Toasters Can you check for #126 to be properly handled by meson? TIA.

**Comment by The-King-of-Toasters** [comment]
Thanks to Ninja you get header checking for free. Touching `CRT.c` only compiles `CRT.o`, but touching `CRT.h` re-compiles everything.

**Comment by BenBE** [comment]
We're not supporting alternative build systems as of now.

**Comment by Salamandar** [comment]
@BenBE do you still refuse this MR ?
Since 2020, Meson became more and more present in the Linux landscape, low-level libs are using it.
Among the arguments, let me cite:
* easy-to-find build options (just read the meson_options.txt file) for maintainers/new users
* out-of-tree build
* out of the box support for advanced build options like LTO, unity build etc.

If you'd reconsider this, I'd love to pick this MR where it was left.

**Comment by fasterit** [comment]
@Salamandar: We still see no reason to support alternative build systems. They all have upsides and downsides and none are compelling enough to consider anything beyond Autotools for now.

**Comment by eli-schwartz** [comment]
> Among the arguments, let me cite:
> 
>  * easy-to-find build options (just read the meson_options.txt file) for maintainers/new users
> 
>  * out-of-tree build
> 
>  * out of the box support for advanced build options like LTO, unity build etc.

To be fair to autotools, it already supports the first two.

All build options are available in the `./configure --help` output (just like `meson configure`). This actually should not be hard for a build system to do, but cmake still fails at it... :D 

autotools does support out of tree builds as well as in-tree builds. This assumes a correctly-written Makefile.am, which is not actually guaranteed since most people don't really check that (if you use `make distcheck`, that will automatically check that VPATH builds succeed, but many people do not distcheck).

Mixing out of tree and in-tree builds on the same sourcedir is a recipe for disaster and will bomb out. Meson simply refuses to support in-tree builds at all.

**Comment by fasterit** [comment]
Autotools will complain as well, if you mix the two: `configure: error: source directory already configured; run "make distclean" there first`.
And FTR: htop builds fine out-of-tree.

I think this is a case of "but I like $otherbuildsystem better". That is fair and fine. Just not reason for us to put any work in.
Autotools is old and not beautiful in any way. But it still works well enough.

**Comment by Salamandar** [comment]
Alright, understood.

**Comment by eli-schwartz** [comment]
> Autotools will complain as well, if you mix the two: `configure: error: source directory already configured; run "make distclean" there first`.

Only if you build in tree first, I think.

> And FTR: htop builds fine out-of-tree.

That is always good to hear. :)

**Pull request closed without merging**